lametro/wagtail_hooks.py

```diff
@@ -23,22 +38,189 @@ class AlertAdmin(ModelAdmin):
+    )
+    organization = django_filters.ModelChoiceFilter(
+        label="Member of",
+        queryset=LAMetroOrganization.objects.filter(memberships__isnull=False)
+        .order_by("name")
+        .distinct(),
+        method="filter_organization",
+    )
+
+    def filter_current_member(self, queryset, name, value):
+        people = [
+            page.person
+            for page in queryset.select_related("person")
+            .prefetch_related("person__memberships")
+            .distinct()
+        ]
+
+        if value == "true":
+            filter_func = lambda person: person.current_memberships.exists()  # noqa
+        else:
+            filter_func = lambda person: not person.current_memberships.exists()  # noqa
+
+        return queryset.filter(
+            person__id__in=(p.id for p in filter(filter_func, people))
+        )
+
+    def filter_organization(self, queryset, name, value):
+        return queryset.filter(person__memberships__organization=value).distinct()
+
+    class Meta:
+        model = BoardMemberDetails
+        fields = ["current_member"]
+
+
+class LinkedStatusTagColumn(LiveStatusTagColumn):
+    cell_template_name = "snippets/related_object_status_tag.html"
+
+    def get_cell_context_data(self, instance, parent_context):
+        print("getting context")
+        context = super().get_cell_context_data(instance, parent_context)
+        context["url"] = instance.get_url()
+        return context
+
+
+class BoardMemberDetailsViewSet(SnippetViewSet):
+    model = BoardMemberDetails
+    icon = "user"
+    add_to_admin_menu = True
+    menu_icon = "user"
+    menu_order = 200
+    filterset_class = BoardMemberFilterSet
+    ordering = ("person__name",)
+    search_fields = ("person__name",)
+    list_display = [
+        "__str__",
+        UpdatedAtColumn(),
+        LinkedStatusTagColumn(),
+    ]
+
+    @property
+    def permission_policy(self):
+        return NoAddModelPermissionPolicy(self.model)
+
+    edit_handler = ObjectList(
+        [
+            HelpPanel(
+                content=(
+                    "<p>On this page, you can manage a board member's headshot and "
+                    "bio. All other details and relationships, e.g., memberships, "
+                    "should be managed in Legistar.</p>"
+                    "<p><strong>Note:</strong> The page preview excludes "
+                    "the map and committee and board report modules displayed "
+                    "on the live page.</p>"
+                )
+            ),
+            MultiFieldPanel(
+                [
+                    FieldPanel("headshot", heading="Image"),
+                    FieldPanel("headshot_source", heading="Source"),
+                ],
+                heading="Headshot",
+            ),
+            FieldPanel("bio"),
+            PublishingPanel(),
+        ]
+    )
+
+
 modeladmin_register(AlertAdmin)
+register_snippet(BoardMemberDetailsViewSet)
+
+
+class UserBarLink:
```

Comment:
I pulled some reusable things out of the model admin link so I could also use them for the snippet edit link.